app/src/test/java/org/mozilla/focus/browser/WebViewCacheTest.kt

```diff
@@ -0,0 +1,46 @@
+package org.mozilla.focus.browser
+
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import mozilla.components.browser.engine.system.SystemEngineView
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.mock
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+class WebViewCacheTest {
+
+    @Test
+    fun `WHEN getWebView is called multiple times THEN the same WebView is returned`() {
+        val cachRefs = List(5) { getWebView() }
+        cachRefs.forEach { assertEquals(cachRefs.first(), it) }
+    }
+
+    @Test
+    fun `GIVEN cached WebView has a parent WHEN WebView is returned THEN WebView should be removed from parent`() {
+        for (i in 0..5) {
```

Comment:
nit: I think this test would make more sense if it was written without a loop: in order to read the core existence of the test (removed from parent), you need to read the bottom assertion and go back to the top assertion.

We don't need to test this more than once or twice so I don't think the loop is necessary.